helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -108,18 +111,19 @@ public String getObjName(ExternalView obj) {
 
   public synchronized void refresh(HelixDataAccessor accessor) {
     long startTime = System.currentTimeMillis();
+    Set<HelixConstants.ChangeType> propertyRefreshed = new HashSet<>();
+
+    // Refresh base
+    super.refresh(accessor, propertyRefreshed);
```

Comment:
minor: why not return the propertyRefreshed from the refresh() call? Pass a parameter and expect it being changed in the method is a bit not-clear.

Reply:
+1

Reply:
I was thinking about keeping the new protected method the same as the original one. But on my second thought, I think it is doable, and not difficult. Let me have a try.
Good point.